galley/pkg/runtime/projections/serviceentry/handler.go

```diff
@@ -261,26 +266,34 @@ func (p *Handler) versionString() string {
 	return fmt.Sprintf("%d", p.version)
 }
 
-func (p *Handler) toMcpResource(svcEntry resource.Entry, endpoints *coreV1.Endpoints) (*mcp.Resource, bool) {
-	fullName := svcEntry.ID.FullName
-	svc := svcEntry.Item.(*coreV1.ServiceSpec)
-	svcMeta := svcEntry.Metadata
-	se := convert.Service(svc, svcMeta, svcEntry.ID.FullName, p.domainSuffix)
+func (p *Handler) newServiceEntry(serviceResource resource.Entry, endpoints *coreV1.Endpoints) *networking.ServiceEntry {
+	se := &networking.ServiceEntry{}
```

Comment:
Does this get stack allocated or heap allocated? 
(Declaring this as a reference to SE, rather than as a SE(and passing the reference to it) makes me think that it would get heap allocated, but Golang compiler might be advanced enough to detect that the reference is not escaping scope).

Reply:
switched to stack allocation. Pass by reference where needed.